components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -369,4 +369,20 @@ public static String prependDomain(String value) {
             return UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME + CarbonConstants.DOMAIN_SEPARATOR + value;
         }
     }
+
+    /**
+     * Method to extract domain name from the input value passed in to this method.
+     *
+     * @param nameWithDomain string which contains domain name
+     * @return extracted domain value or empty string if no domain.
+     */
+    public static String extractDomain(String nameWithDomain) {
+
+        if (nameWithDomain != null && nameWithDomain.indexOf("/") > 0) {
```

Comment:
Can we use domain separator
https://github.com/wso2/carbon-kernel/blob/949e17f14eab27f857800eb8f5064d67432b19f1/core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/CarbonConstants.java#L388-L397

Reply:
Fixed with 1aa2ee1877d0db55c7f278d5395b0ce198485f0c